source/agora/common/API.d

```diff
@@ -47,7 +47,23 @@ public struct PublicConfig
     public bool is_validator;
 }
 
-/// API
+/*******************************************************************************
+
+    Define the API a full node exposes to the world
+
+    A full node:
+    - Can connect to any node and request data about the blockchain & network
+    - Accepts external connections and send them blockchain/network data
+    - Store the data it receives on disk
+    - Can catch up with the network when found lagging behind
+    - Validates the data it receives
+    - Receives, stores and forwards transactions (but drop them after a timeout)
+    - Does not participate in consensus
+
```

Comment:
- Check the node version.

Reply:
Provide node performance score.

Reply:
I think we can add these later? There is a lot of things the node will do, like verifying authenticity of signatures, etc. But we should list these once we add them later. cc @Geod24

Reply:
I tried to focus on blockchain-related stuff. Node performance is not part of those, but the node version might be a good one.

Reply:
- HTTP traffic from the outside world

Reply:
- Full ledger provided

Reply:
- Listening and propagating their decisions blindly to anyone who asks.